.github/workflows/android_build_ubdiag_upload.yml

```diff
@@ -0,0 +1,108 @@
+name: Build and Upload to UBDiag
+
+on:
+  push:
+  workflow_call:
+    inputs:
+      projectKey:
+        type: string
+        required: true
+      app:
+        type: string
+        required: true
+      flavor:
+        type: string
+        required: true
+      appModuleDirectory:
+        type: string
+        required: false
+        default: './app'
+    secrets:
+      ANDROID_JENKINS_PAT:
+        required: true
+      UB_ARTIFACTORY_URL_ANDROID:
+        required: true
+      UB_ARTIFACTORY_USERNAME:
+        required: true
+      UB_ARTIFACTORY_PASSWORD:
+        required: true
+      UBIQUE_POEDITOR_API_KEY:
+        required: true
+      UBDIAG_UPLOAD_URL:
+        required: true
+      ADDITIONAL_GRADLE_PROPS:
+        required: false
+
+jobs:
+  build_dev:
+    name: Build ${{ inputs.flavor }} Flavor
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout repository and submodules
+      - name: Checkout
+        uses: actions/checkout@v2.3.5
+        with:
+          token: ${{ secrets.ANDROID_JENKINS_PAT }}
+          submodules: 'recursive'
+
+      # Set build variables for reuse in multiple steps
+      - name: Set Build Variables
+        id: vars
+        run: |
+          flavor=${{ inputs.flavor }}
+          echo ::set-output name=build_uuid::$(cat /proc/sys/kernel/random/uuid)
+          echo ::set-output name=web_icon::${{ github.workspace }}/app/tmp_icon_large_for_backend.png
+          echo ::set-output name=flavor_capitalized::${flavor~}
+
+      # Setup the build environment with Java 11 and the Zulu OpenJDK
+      - name: Setup Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+
+      # Setup the build environment with Gradle
+      - name: Build app
+        uses: gradle/gradle-build-action@v2
+        with:
+          # Cache entries are not shared among branches. To reduce the amount of cache writes, only allow the main branch to write cache entries and let other branches read them
+          # https://github.com/gradle/gradle-build-action#only-write-to-the-cache-from-the-default-branch
+          cache-read-only: ${{ github.ref_name != 'main' }}
+          arguments: |
+            assemble${{ steps.vars.outputs.flavor_capitalized }}Release
```

Comment:
Ein Problem dass wir haben, ist dass hier ImageMagick nicht (?) installiert ist, was für das Dev-Banner benötigt wird.

Reply:
Gemäss https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#administrative-privileges-of-github-hosted-runners sollten wir das Package einfach installieren können in einem vorherigen Step

Reply:
Imagemagick ist eigentlich schon installiert: https://github.com/UbiqueInnovation/swisstopo-android/runs/5096541104?check_suite_focus=true, siehe Step "Install Imagemagick":
`imagemagick is already the newest version (8:6.9.10.23+dfsg-2.1ubuntu11.4).`